### PR TITLE
Overhaul bootmagic logic to have single entrypoint

### PR DESCRIPTION
**Pull request by zvecr**
## Description

* migrate to quantum
* remove 2 entry points for full bootmagic vs bootmagic lite
* remove some duplication between magic and full bootmagic

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: common_features.mk [hunk 1]
```diff
@@ -318,10 +318,6 @@ ifeq ($(strip $(TERMINAL_ENABLE)), yes)
     OPT_DEFS += -DUSER_PRINT
 endif
 
-ifeq ($(strip $(USB_HID_ENABLE)), yes)
```
Comment (zvecr) [inline common_features.mk]: This had to be moved as `usb_hid.mk` seemed to be trashing the VPATH, its also protocol logic so more suited to `protocol.mk` anyway.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

File: quantum/bootmagic/bootmagic_full.c [hunk 2]
```diff
@@ -117,10 +145,6 @@ void bootmagic(void) {
     }
     if (default_layer) {
         eeconfig_update_default_layer(default_layer);
-        default_layer_set((layer_state_t)default_layer);
-    } else {
-        default_layer = eeconfig_read_default_layer();
-        default_layer_set((layer_state_t)default_layer);
     }
     /* Also initialize layer state to trigger callback functions for layer_state */
     layer_state_set_kb((layer_state_t)layer_state);
```
Comment (drashna) [inline quantum/bootmagic/bootmagic_full.c:150]: Thisline should be moved into magic, with the default_layer_set call.

**Comment by drashna** [comment]
Was this ready to merge?

**Comment by zvecr** [comment]
Think so, was just giving it another test now i point to develop. Still seems to function correctly.

**Pull request merged**